project/driver_controls/src/main.c

```diff
@@ -0,0 +1,116 @@
+#include <stdio.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "input_interrupt.h"
+#include "driver_state.h"
+#include "soft_timer.h"
+#include "driver_devices.h"
+
+#include "power_state.h"
+#include "pedal_state.h"
+#include "direction_state.h"
+#include "turn_signal_state.h"
+#include "hazard_light_state.h"
+#include "mechanical_brake.h"
+
+#define INPUT_DEVICES 16
+#define OUTPUT_DEVICES 1
+
+// The struct of FSMs to be used in the system
+typedef struct FSMGroup {
+  FSM power;
+  FSM pedal;
+  FSM direction;
+  FSM turn_signal;
+  FSM hazard_light;
+  FSM mechanical_brake;
+} FSMGroup;
+
+void device_init() {
+  driver_controls_init();
+
+  Device inputs[INPUT_DEVICES] = {
```

Comment:
Please use designated initializers. This is pretty hard to understand and I don't have any context as to where these values are coming from.